apps/web/src/pages/integrations/CreateProviderPage.tsx

```diff
@@ -8,8 +8,10 @@ export function CreateProviderPage() {
   const navigate = useNavigate();
   const location = useLocation();
 
-  const onIntegrationCreated = (integrationId: string) => {
-    navigate(`/integrations/${integrationId}`);
+  const onIntegrationCreated = (integrationId: string, hasSimilarIntegration: boolean) => {
+    navigate(`/integrations/${integrationId}`, {
+      state: { hasSelectPrimaryProvider: hasSimilarIntegration },
+    });
```

Comment:
when navigating from the create sidebar page to the update sidebar we do set the location state `hasSelectPrimaryProvider` which is used to show the "Select Primary Provider Modal"